src/uart/registers.rs

```diff
@@ -0,0 +1,507 @@
+    /// - 1: When FIFO level `>=` UART_CTRL.rx_thd_val, RTS is deasserted
+    set_hardware_flow_rts_deassert_condition}
+
+    bit_impl! {13, RW,
+    /// # Hardware Flow Control
+    /// Choose if hardware flow control is enabled, or disabled
+    /// - 0: Disabled
+    /// - 1: Enabled
+    set_hardware_flow_control,
+    /// # Check Hardware Flow Control
+    /// Checks if hardware flow control is enabled
+    /// - 0: Disabled
+    /// - 1: Enabled
+    check_hardware_flow_control}
+
+    bit_impl! {12, RW,
+    /// # Number of stop bits
+    /// Set the number of stop bits
+    /// - 0: 1 stop bit
+    /// - 1: 1.5 stop bit for 5 bit mode, 2 bit mode otherwise
+    set_number_of_stop_bits,
+    /// # Check Number Of Stop Bits
+    /// Check the current number of stop bits
+    /// - 0: 1 stop bit
+    /// - 1: 1.5 stop bit for 5 bit mode, 2 bit mode otherwise
+    check_number_of_stop_bits}
+
+    bit_impl! {10..=11, RW u8,
+    /// # Set Character Length
+    /// Set the number of bits in a character in an UART frame.
+    /// - 0: 5 bits
+    /// - 1: 6 bits
+    /// - 2: 7 bits
+    /// - 3: 8 bits
+    set_character_length,
+    /// # Check Character Length
+    /// Checks the current number of bits in a character in a UART frame
+    /// - 0: 5 bits
+    /// - 1: 6 bits
+    /// - 2: 7 bits
+    /// - 3: 8 bits
+    check_character_length}
+
+    bit_impl! {9, RESET, // FIXME This needs to be renamed / changed
+    /// # Activate Receive FIFO Flush
+    /// Write a 1 to flush the receive FIFO
+    activate_receive_fifo_flush}
+
+    bit_impl! {8, RESET, // FIXME This need to be renamed / changed
+    /// # Activate Transmit FIFO Flush
+    /// Write a 1 to flush the transmit FIFO
+    activiate_transmit_fifo_flush}
+
+    bit_impl! {7, RW,
+    /// # Set `CTS` Sampling Disable
+    /// Choose to enable or disable `CTS` (Clear To Send)
+    /// - 0: Enabled
+    /// - 1: Disabled
+    set_cts_sampling_disable,
+    /// # Check `CTS` Sampling Disable
+    /// Check the current state of CT disable
+    /// - 0: Enabled
+    /// - 1: Disabled
+    check_cts_sampling_disable}
+
+    bit_impl! {6, RW,
+    /// # Set Parity Value
+    set_parity_value,
+    /// # Check Parity Value
+    check_parity_value}
+
+    bit_impl! {5, RW,
+    /// # Parity Odd Even Select
+    set_parity_odd_even,
+    /// # Check Parity Odd Even Select
+    check_parity_odd_even}
+
+    bit_impl! {4, RW,
+    /// # Set Transmit Parity Generation Enable
+    set_transmit_parity_generation_enable,
+    /// # Check Transmit Parity Generation Enable
+    check_transmit_parity_genration_enable}
+
+    bit_impl! {0..=3, RW u8,
+    /// # Set Receive FIFO Threshold
+    set_recieve_fifo_threshold,
+    /// # Check Receive FIFO Threshold
+    check_recieve_fifo_threshold}
+}
+
+/// # UART Status Register
+/// The UART Status Register. See page 182, table 12-9
+pub struct StatusRegister<const PORT_PTR: usize> {}
+reg_impl!(RO, StatusRegister, uro::UART_CTRL);
```

Comment:
`uro::UART_CTRL` constant should only be used for the `ControlRegister` struct. The `StatusRegister` struct should a different one.